enginetest/enginetests.go

```diff
@@ -1686,9 +1682,31 @@ func TestVersionedViews(t *testing.T, harness Harness) {
 	require.NoError(err)
 	iter.Close(ctx)
 
+	// views with unions
+	_, iter, err = e.Query(ctx, "CREATE VIEW myview3 AS SELECT i from myview1 union select s from myhistorytable")
+	require.NoError(err)
+	iter.Close(ctx)
+
+	// views with subqueries
+	_, iter, err = e.Query(ctx, "CREATE VIEW myview4 AS SELECT * FROM myhistorytable where i in (select distinct cast(RIGHT(s, 1) as signed) from myhistorytable)")
```

Comment:
Think you need one more for completeness, something like:

`CREATE VIEW myview5 AS SELECT * FROM (select * from myhistorytable where i in (select distinct cast(RIGHT(s, 1) as 
signed)) as sq`

This will handle the SubqueryAlias case, which I believe is missing.